### PR TITLE
ir+lower: remove old TIR `stores()` API and migrate to new methods

### PR DESCRIPTION
**Pull request by feds01**

### CODE REVIEW

File: compiler/hash-lower/src/build/into.rs [hunk 1]
```diff
@@ -193,16 +192,13 @@ impl<'tcx> BodyBuilder<'tcx> {
...
-                    FnCallTermKind::LogicalBinOp(op, _, _) => {
+                    FnCallTermKind::LogicalBinOp(op, lhs_term, rhs_term) => {
                         let (short_circuiting_block, mut else_block, join_block) = (
                             self.control_flow_graph.start_new_block(),
                             self.control_flow_graph.start_new_block(),
                             self.control_flow_graph.start_new_block(),
                         );
 
-                        let lhs_term = self.stores().args().get_at_index(*args, 1).value;
-                        let rhs_term = self.stores().args().get_at_index(*args, 2).value;
```
Comment (feds01) [inline compiler/hash-lower/src/build/into.rs:204]: Might be responsible for different IR gen?

File: compiler/hash-lower/src/build/rvalue.rs [hunk 2]
```diff
@@ -23,41 +23,39 @@ use crate::build::category::RValueKind;
 
 impl<'tcx> BodyBuilder<'tcx> {
     /// Construct an [RValue] from the given [ast::Expr].
-    pub(crate) fn as_rvalue(&mut self, mut block: BasicBlock, term_id: TermId) -> BlockAnd<RValue> {
-        // @@Temporary: replace with get_ref();
-        let term = self.stores().term().get(term_id);
-        let span = self.span_of_term(term_id);
+    pub(crate) fn as_rvalue(&mut self, mut block: BasicBlock, term: TermId) -> BlockAnd<RValue> {
```
Comment (feds01) [inline compiler/hash-lower/src/build/rvalue.rs:26]: rename `term` to `rvalue` for clarity

File: compiler/hash-lower/src/build/into.rs (hunk 1 above)
Comment (kontheocharis) [reply compiler/hash-lower/src/build/into.rs:204]: Not sure what you mean

Comment (feds01) [reply compiler/hash-lower/src/build/into.rs:204]: Im investigating why ir changed, its a self comment

Comment (feds01) [reply compiler/hash-lower/src/build/into.rs:204]: Yeah, this is actually a bug in lowering that I accidentally discovered, it should be that I need to get the `2` and `3` arguments.

File: tests/cases/lowering/early_returns.stdout [hunk 3]
```diff
@@ -9,9 +9,10 @@ boo := (_1: i32, _2: i32) -> i32 {
     mut _3: bool;
     mut _4: bool;
     mut _5: bool;
-    mut _6: i32;
+    mut _6: bool;
```
Comment (kontheocharis) [inline tests/cases/lowering/early_returns.stdout:12]: I'm guessing this is the difference?

**Review by kontheocharis — APPROVED** [review]

**Pull request merged**